recipes/coin-lemon/all/conandata.yml

```diff
@@ -6,3 +6,9 @@ patches:
   "1.3.1":
     - patch_file: "patches/cmake-add-runtime-destination_1.3.1.patch"
     - patch_file: "patches/0001-remove-register-keyword.patch"
+    - patch_file: "patches/fix_cpp_17_compilation_warnings.patch"
+      patch_type: "portability"
+      patch_description: "Fix C++17 compilation warnings regarding the use of the deprecated std::iterator"
```

Comment:
https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#rules

I am not sure the warnings deserve a patch. These are costly to maintain and we really want ones to make "working software" 

Does this fail to build with later cppstd? I would prefer less patches otherwise

Reply:
Hi @prince-chrismc ,

I'm more in favour of a 'treatiing warnings as errors' policy. Anyway, yes, some of these warnings cause failures when compiling with C++17 or newer (see #[15515](https://github.com/conan-io/conan-center-index/issues/15515) ).

Best regards,
Roberto.